synapse/rest/client/v2_alpha/keys.py

```diff
@@ -51,23 +54,47 @@ class KeyUploadServlet(RestServlet):
       },
     }
     """
-    PATTERNS = client_v2_patterns("/keys/upload/(?P<device_id>[^/]*)", releases=())
+    PATTERNS = client_v2_patterns("/keys/upload(/(?P<device_id>[^/]+))?$",
+                                  releases=(), v2_alpha=False)
```

Comment:
Is Vector the only thing using that API? I though Tor had got a client using the e2e stuff.